common/djangoapps/embargo/models.py

```diff
@@ -72,6 +82,213 @@ def embargoed_countries_list(self):
+        db_index=True, unique=True,
+        help_text=ugettext_lazy(u"Two character ISO country code.")
+    )
+
+    def __unicode__(self):
+        return u"{name} ({code})".format(
+            name=unicode(self.country.name),
+            code=unicode(self.country)
+        )
+
+    class Meta:
+        # Default ordering is ascending by country code
+        ordering = ['country']
+
+
+class CountryAccessRule(models.Model):
+    """Course access rule based on the user's country.
+
+    The rule applies to a particular course-country pair.
+    Countries can either be whitelisted or blacklisted,
+    but not both.
+
+    To determine whether a user has access to a course
+    based on the user's country:
+
+    1) Retrieve the list of whitelisted countries for the course.
+    (If there aren't any, then include every possible country.)
+
+    2) From the initial list, remove all blacklisted countries
+    for the course.
+
+    """
+
+    RULE_TYPE_CHOICES = (
+        (WHITE_LIST, 'Whitelist (allow only these countries)'),
+        (BLACK_LIST, 'Blacklist (block these countries)'),
+    )
+
+    rule_type = models.CharField(
+        max_length=255,
+        choices=RULE_TYPE_CHOICES,
+        default=BLACK_LIST,
+        help_text=ugettext_lazy(
+            u"Whether to include or exclude the given course. "
+            u"If whitelist countries are specified, then ONLY users from whitelisted countries "
+            u"will be able to access the course.  If blacklist countries are specified, then "
+            u"users from blacklisted countries will NOT be able to access the course."
+        )
+    )
+
+    restricted_course = models.ForeignKey(
+        "RestrictedCourse",
+        help_text=ugettext_lazy(u"The course to which this rule applies.")
+    )
+
+    country = models.ForeignKey(
+        "Country",
+        help_text=ugettext_lazy(u"The country to which this rule applies.")
+    )
+
+    @classmethod
+    def is_course_embargoed_in_country_list(cls, course_id, country, rule_type):
+        """
+        Check if the country is either in whitelist or blacklist of countries for the course_id
+
+        Args:
+            course_id (str): course_id to look for
+            country (str): A 2 characters code of country
+            rule_type (str): whitelist or blacklist
+
+        Returns:
+            Boolean
+            True if no country found for the given course and rule type
+            otherwise check given country exists in list
+        """
+        country_access_rules = cls._get_course_embargoed_list_countries_with_rule_type(course_id, rule_type)
+        if not country_access_rules:
+            return True
+
+        return country in country_access_rules
+
+
+    @classmethod
+    def cache_key_name(cls, course_id, cache_type):
+        return "{}/embargo/countries/{}".format(course_id, cache_type)
+
+    @classmethod
+    def _get_course_embargoed_list_countries_with_rule_type(cls, course_id, rule_type):
```

Comment:
Nitpick: I'd call this `_get_country_access_rules_for_course(course_id, rule_type)`.  "Embargo" means that a country has imposed trade sanctions on another country, which is true for some -- but not all -- of the use cases we need to support.